src/Domain.php

```diff
@@ -419,10 +419,43 @@ public static function showMassiveActionsSubForm(MassiveAction $ma)
 
         switch ($ma->getAction()) {
             case 'add_item':
+                Dropdown::show(
+                    'DomainRelation',
+                    [
+                        'name'   => "domainrelations_id",
+                        'value'  => DomainRelation::BELONGS,
+                        'display_emptychoice'   => false
+                    ]
+                );
                 self::dropdownDomains([]);
                 echo "&nbsp;" .
                  Html::submit(_x('button', 'Post'), ['name' => 'massiveaction']);
                 return true;
+            case 'remove_domain':
+                $compselected = $ma->getInput()['items']['Computer'];
+                $used = [];
+                //Display only the domain used by a computer
+                $domain_item = new Domain_Item();
+                $domain = new self();
+                foreach ($domain->find() as $key) {
+                    $used[] = $key['id'];
+                    foreach (
+                        $domain_item->find([
+                            'itemtype' => 'Computer',
+                            'items_id' => $compselected
+                        ]) as $domain_used
+                    ) {
+                        if (in_array($domain_used['domains_id'], $used)) {
+                            unset($used[array_search($domain_used['domains_id'], $used)]);
+                        }
+                    }
+                }
```

Comment:
On a GLPI instance with many domains and many computers, it would generate lots of queries. I think it could be done in a unique request, using a subquery.
```php
global $DB;
$domains_iterator = $DB->request([
    'SELECT' => ['id'],
    'FROM'   => 'glpi_domains',
    'WHERE'  => [
        'id' => new QuerySubQuery([
            'SELECT'   => 'domains_id',
            'DISTINCT' => true,
            'FROM'     => 'glpi_domains_items',
            'WHERE'    => [
                'itemtype' => 'Computer',
                'items_id' => $compselected
            ]
        ])
    ]
);
```

Reply:
or not to filter this list, 
if the user chooses a non-attached domain,
GLPI will do nothing

Reply:
@cedric-anne what do you think?

Reply:
On massive actions, it can make sense to not filter anything.